extensions/vi-mode/visual.lisp

```diff
@@ -248,3 +248,30 @@
         (move-to-column *start-point* end-col)
         (move-to-column (current-point) start-col))
       (vi-visual-swap-points)))
+
+(defmethod lem/language-mode:set-region-point ((start point) (end point) (global-mode (eql :|vi|)))
+  (declare (ignore global-mode))
+  (when (visual-p)
+    (let ((v-range (visual-range)))
+      (move-point start (car v-range))
+      (move-point end (cadr v-range)))))
+
+
+(eval-when (:compile-toplevel :load-toplevel)
+  #+sbcl
+  (sb-ext:unlock-package :lem-base)
```

Comment:
I cannot accept this kind of barbaric behavior.

Reply:
Fair, I also think it's kind of ugly, but I would have to change a lot of how lem is written to accommodate an improvement approach. I'll dig around and see what I can do, thanks for the feedback :+1: